### PR TITLE
all: change optional to result in most of the libraries

### PR DESCRIPTION
**Pull request by yuyi98**
This PR change optional to result in most of the libraries.

### CODE REVIEW

**Comment by spytheman** [comment]
> @spytheman Sorry, we still need to merge this PR first, and then fix vls and other applications.

ok, I think it will be worth it.

**Review by spytheman — APPROVED** [review]

**Comment by spytheman** [comment]
`coreutils` seems ok with v/vlib from this PR

**Comment by spytheman** [comment]
`vls` needed only cosmetic changes, I'll push a PR for it soon.

**Comment by spytheman** [comment]
`vab` compiles cleanly

**Comment by spytheman** [comment]
`gitly` compiles cleanly

**Comment by spytheman** [comment]
`vlang/pdf` compiles cleanly as well.

**Comment by spytheman** [comment]
Latest `ui` fb83c20 compiles cleanly too (it needed just a small cosmetic change).

**Comment by spytheman** [comment]
VTL and VSL also compile cleanly in their latest version.

**Comment by spytheman** [comment]
TLDR: 

These 2 should be merged right after this PR is merged, the rest were either already OK, are are now fixed:
https://github.com/vlang/go2v/pull/98
https://github.com/vlang/vls/pull/399

The PR for VEX should be merged too, or we would need to comment out the check for it temporarily in this PR (I do not have rights to merge there, imho only Ned has, so I've asked him to do it):
https://github.com/nedpals/vex/pull/58

**Comment by spytheman** [comment]
Everything should be OK now (VEX is skipped temporarily on the CI).

**Comment by yuyi98** [comment]
@spytheman  Thank you for making a lot of corrections !!!
